pkg/storage/stores/indexshipper/compactor/retention/retention.go

```diff
@@ -204,7 +218,13 @@ func markforDelete(ctx context.Context, tableName string, marker MarkerStorageWr
 		return false, nil
 	})
 	if err != nil {
-		return false, false, err
+		if errors.Is(err, context.DeadlineExceeded) && errors.Is(iterCtx.Err(), context.DeadlineExceeded) {
+			// Deletes timed out. Don't return an error so compaction can continue and deletes can be retried
+			level.Warn(util_log.Logger).Log("msg", "Timed out while running delete")
```

Comment:
Does this already have the userID and or delete request ID (perhaps using a custom logger)? If not please add it.